distributed/worker.py

```diff
@@ -2947,21 +2947,36 @@ async def execute(self, key, report=False):
             )  # TODO: comment out?
             assert key == ts.key
             try:
-                result = await self.executor_submit(
-                    ts.key,
-                    apply_function,
-                    args=(
-                        function,
-                        args2,
-                        kwargs2,
-                        self.execution_state,
+                e = self.executors[executor]
+                if "ThreadPoolExecutor" in str(type(e)):
```

Comment:
Why not an `isinstance` check?

Reply:
We've forked the `concurrent.futures.ThreadPoolExecutor` internally to support secede/rejoin.  I would want this check to work with either ours or the standard module's

Reply:
``isinstance(inst, (cls1, cls2))``?

Reply:
Loky is another example of someone implementing something like this (although they only do ProcessPoolExecutor).

To me this choice feels a bit ugly, but also ok?